activerecord/lib/active_record/associations.rb

```diff
@@ -1357,6 +1358,9 @@ module ClassMethods
         #   association objects.
         # [:strict_loading]
         #   Enforces strict loading every time the associated record is loaded through this association.
+        # [:owner_ensuring_destroy]
```

Comment:
I'm not happy with this option name. Any suggestions?

Reply:
Do we have a naming convention for options that refer to method names?  If not, is one of these preferable:

* `ensure_owner_destroyed: :deleted?`
* `ensures_owner_destroyed: :deleted?`
* `ensure_owner_destroyed_with: :deleted?`

Also, would `verify` be better than `ensure`, since `ensure` sounds like it will make it happen if it is not already done?

* `verify_owner_destroyed_with: :deleted?`

Reply:
`verify_owner_destroyed_with` is a good name. Thank you for the suggestion @jonathanhefner

Reply:
> Also, would `verify` be better than `ensure`, since `ensure` sounds like it will make it happen if it is not already done?

`ensure` is meant as a prerequisite, like in `destroy_later ensuring: :due?`. I read it as "can't proceed if this hasn't happened".

And I do think there's something nice about reusing the wording, though perhaps it's also confusing because it's targeting the owner and not the associated records.

My take would be:

```ruby
has_many :comments, dependent: :destroy_later, ensuring_owner_was: :destroyed?
```

`_was` to imply the check is run in the job i.e. later, `ensuring` to match the prerequisite option on `record.destroy_later` also has, and `owner` to indicate the check is on the owner of course. A lot of the meaning is filled in via the supplied predicate granted, but that seems ok for the option to me.